File: platform/frontend/src/api/frameworkconfig.ts
import { authFetch } from './auth'

const API = import.meta.env.VITE_API_URL || ''

export interface FrameworkConfig {
  selectedFramework: string
  selectedBackend: string
  selectedDatabase: string
  selectedStyling: string
  projectsGenerated: number
  autoDetectStack: boolean
  includeDocker: boolean
  includeCI: boolean
  includeTests: boolean
  favoriteFrameworks: string[]
}

export interface Framework {
  id: string
  name: string
  category: string
  tier: number
  language: string
  description: string
  icon: string
  tags: string[]
}

export interface BackendOption {
  id: string
  name: string
  description: string
}

export interface DatabaseOption {
  id: string
  name: string
  description: string
}

export interface GeneratePreviewResult {
  framework: string
  projectStructure: string[]
  estimatedFiles: number
  estimatedTokens: number
  estimatedCost: number
  estimatedTimeSeconds: number
  features: string[]
}

export interface FrameworkStats {
  totalProjects: number
  favoriteFramework: string
  frameworksUsed: number
  recentProjects: { framework: string; projectName: string; generatedAt: string }[]
}

export async function getFrameworkConfig(): Promise<FrameworkConfig> {
  const res = await authFetch(`${API}/api/framework/config`)
  if (!res.ok) throw new Error('Failed to load framework config')
  return res.json()
}

export async function updateFrameworkConfig(data: Partial<FrameworkConfig>): Promise<{ success: boolean }> {
  const res = await authFetch(`${API}/api/framework/config`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  })
  if (!res.ok) throw new Error('Failed to update config')
  return res.json()
}

export async function getFrameworks(): Promise<Framework[]> {
  const res = await authFetch(`${API}/api/framework/frameworks`)
  if (!res.ok) throw new Error('Failed to load frameworks')
  return res.json()
}

export async function getBackends(): Promise<BackendOption[]> {
  const res = await authFetch(`${API}/api/framework/backends`)
  if (!res.ok) throw new Error('Failed to load backends')
  return res.json()
}

export async function getDatabases(): Promise<DatabaseOption[]> {
  const res = await authFetch(`${API}/api/framework/databases`)
  if (!res.ok) throw new Error('Failed to load databases')
  return res.json()
}

export async function generatePreview(framework?: string, description?: string): Promise<GeneratePreviewResult> {
  const res = await authFetch(`${API}/api/framework/generate-preview`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ framework, description }),
  })
  if (!res.ok) throw new Error('Failed to generate preview')
  return res.json()
}

export async function getFrameworkStats(): Promise<FrameworkStats> {
  const res = await authFetch(`${API}/api/framework/stats`)
  if (!res.ok) throw new Error('Failed to load stats')
  return res.json()
}

File: platform/backend/AiDevRequest.API/Entities/FrameworkConfig.cs
using System.ComponentModel.DataAnnotations;

namespace AiDevRequest.API.Entities;

public class FrameworkConfig
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public string UserId { get; set; } = "";

    public string SelectedFramework { get; set; } = "react-vite";
    public string SelectedBackend { get; set; } = "none";
    public string SelectedDatabase { get; set; } = "none";
    public string SelectedStyling { get; set; } = "tailwind";

    public int ProjectsGenerated { get; set; } = 0;
    public string LastGeneratedProjectId { get; set; } = "";
    public string FavoriteFrameworks { get; set; } = "";

    public string CustomTemplateJson { get; set; } = "{}";
    public string FrameworkHistoryJson { get; set; } = "[]";

    public bool AutoDetectStack { get; set; } = true;
    public bool IncludeDocker { get; set; } = false;
    public bool IncludeCI { get; set; } = false;
    public bool IncludeTests { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
